### PR TITLE
docs: code review etiquette

### PR DESCRIPTION
**Pull request by boweihan**
# Description

This PR adds a short list of guidelines for etiquette surrounding code reviews.

## Type of change

- [x] Developer documentation

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

**Comment by boweihan** [comment]
Opinions welcome :)

File: CODE_REVIEW.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## Authoring
+
+- Concisely describe the purpose the pull request `("this is a fix for ...")`
+- Provide an explanation for why the work is being done that can be understood without historical context
+- Describe _why_ your code change serves its intended purposes. How does it fix the bug? Why is this the right refactor?
+- @tag individuals that you specifically want to involve in the review
+
+## Reviewing
+
+- Approach code reviews with the mindset to help others learn and grow
```
Comment (tracy-french) [inline CODE_REVIEW.md:14]: Eh, I don't know about this one. It's too easy to block a CR if the mindset is not objective.

File: CODE_REVIEW.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- Familiarize yourself with the context of the issue and why the PR exists _before_ leaving comments
+- Label comments with intent: `("nit:... blocking:... consider:... personal preference:... ")`
```
Comment (tracy-french) [inline CODE_REVIEW.md:16]: Love this.

File: CODE_REVIEW.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- Ask, don't tell `("what do you think about trying...?")`
```
Comment (tracy-french) [inline CODE_REVIEW.md:17]: Not sure about this. Why add subjectivity? How do you resolve this? What if the author thinks differently?

**Comment by tracy-french** [comment]
Discussed offline. Let's consider adding some of the ideas from Google https://google.github.io/eng-practices/review/reviewer/.

**Comment by tracy-french** [comment]
Playwright test failures should resolve after a rebase.

**Comment by tracy-french** [comment]
@boweihan What are the next steps for this?

**Comment by boweihan** [comment]
Closing this for now.

**Pull request closed without merging**